<!DOCTYPE html>
<html>
<head>
<title>Week_8_Practical-VCF_Analysis.md</title>
<meta http-equiv="Content-type" content="text/html;charset=UTF-8">

<style>
/* https://github.com/microsoft/vscode/blob/master/extensions/markdown-language-features/media/markdown.css */
/*---------------------------------------------------------------------------------------------
 *  Copyright (c) Microsoft Corporation. All rights reserved.
 *  Licensed under the MIT License. See License.txt in the project root for license information.
 *--------------------------------------------------------------------------------------------*/

body {
	font-family: var(--vscode-markdown-font-family, -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif);
	font-size: var(--vscode-markdown-font-size, 14px);
	padding: 0 26px;
	line-height: var(--vscode-markdown-line-height, 22px);
	word-wrap: break-word;
}

#code-csp-warning {
	position: fixed;
	top: 0;
	right: 0;
	color: white;
	margin: 16px;
	text-align: center;
	font-size: 12px;
	font-family: sans-serif;
	background-color:#444444;
	cursor: pointer;
	padding: 6px;
	box-shadow: 1px 1px 1px rgba(0,0,0,.25);
}

#code-csp-warning:hover {
	text-decoration: none;
	background-color:#007acc;
	box-shadow: 2px 2px 2px rgba(0,0,0,.25);
}

body.scrollBeyondLastLine {
	margin-bottom: calc(100vh - 22px);
}

body.showEditorSelection .code-line {
	position: relative;
}

body.showEditorSelection .code-active-line:before,
body.showEditorSelection .code-line:hover:before {
	content: "";
	display: block;
	position: absolute;
	top: 0;
	left: -12px;
	height: 100%;
}

body.showEditorSelection li.code-active-line:before,
body.showEditorSelection li.code-line:hover:before {
	left: -30px;
}

.vscode-light.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(0, 0, 0, 0.15);
}

.vscode-light.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(0, 0, 0, 0.40);
}

.vscode-light.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-dark.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 255, 255, 0.4);
}

.vscode-dark.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 255, 255, 0.60);
}

.vscode-dark.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

.vscode-high-contrast.showEditorSelection .code-active-line:before {
	border-left: 3px solid rgba(255, 160, 0, 0.7);
}

.vscode-high-contrast.showEditorSelection .code-line:hover:before {
	border-left: 3px solid rgba(255, 160, 0, 1);
}

.vscode-high-contrast.showEditorSelection .code-line .code-line:hover:before {
	border-left: none;
}

img {
	max-width: 100%;
	max-height: 100%;
}

a {
	text-decoration: none;
}

a:hover {
	text-decoration: underline;
}

a:focus,
input:focus,
select:focus,
textarea:focus {
	outline: 1px solid -webkit-focus-ring-color;
	outline-offset: -1px;
}

hr {
	border: 0;
	height: 2px;
	border-bottom: 2px solid;
}

h1 {
	padding-bottom: 0.3em;
	line-height: 1.2;
	border-bottom-width: 1px;
	border-bottom-style: solid;
}

h1, h2, h3 {
	font-weight: normal;
}

table {
	border-collapse: collapse;
}

table > thead > tr > th {
	text-align: left;
	border-bottom: 1px solid;
}

table > thead > tr > th,
table > thead > tr > td,
table > tbody > tr > th,
table > tbody > tr > td {
	padding: 5px 10px;
}

table > tbody > tr + tr > td {
	border-top: 1px solid;
}

blockquote {
	margin: 0 7px 0 5px;
	padding: 0 16px 0 10px;
	border-left-width: 5px;
	border-left-style: solid;
}

code {
	font-family: Menlo, Monaco, Consolas, "Droid Sans Mono", "Courier New", monospace, "Droid Sans Fallback";
	font-size: 1em;
	line-height: 1.357em;
}

body.wordWrap pre {
	white-space: pre-wrap;
}

pre:not(.hljs),
pre.hljs code > div {
	padding: 16px;
	border-radius: 3px;
	overflow: auto;
}

pre code {
	color: var(--vscode-editor-foreground);
	tab-size: 4;
}

/** Theming */

.vscode-light pre {
	background-color: rgba(220, 220, 220, 0.4);
}

.vscode-dark pre {
	background-color: rgba(10, 10, 10, 0.4);
}

.vscode-high-contrast pre {
	background-color: rgb(0, 0, 0);
}

.vscode-high-contrast h1 {
	border-color: rgb(0, 0, 0);
}

.vscode-light table > thead > tr > th {
	border-color: rgba(0, 0, 0, 0.69);
}

.vscode-dark table > thead > tr > th {
	border-color: rgba(255, 255, 255, 0.69);
}

.vscode-light h1,
.vscode-light hr,
.vscode-light table > tbody > tr + tr > td {
	border-color: rgba(0, 0, 0, 0.18);
}

.vscode-dark h1,
.vscode-dark hr,
.vscode-dark table > tbody > tr + tr > td {
	border-color: rgba(255, 255, 255, 0.18);
}

</style>

<style>
/* Tomorrow Theme */
/* http://jmblog.github.com/color-themes-for-google-code-highlightjs */
/* Original theme - https://github.com/chriskempson/tomorrow-theme */

/* Tomorrow Comment */
.hljs-comment,
.hljs-quote {
	color: #8e908c;
}

/* Tomorrow Red */
.hljs-variable,
.hljs-template-variable,
.hljs-tag,
.hljs-name,
.hljs-selector-id,
.hljs-selector-class,
.hljs-regexp,
.hljs-deletion {
	color: #c82829;
}

/* Tomorrow Orange */
.hljs-number,
.hljs-built_in,
.hljs-builtin-name,
.hljs-literal,
.hljs-type,
.hljs-params,
.hljs-meta,
.hljs-link {
	color: #f5871f;
}

/* Tomorrow Yellow */
.hljs-attribute {
	color: #eab700;
}

/* Tomorrow Green */
.hljs-string,
.hljs-symbol,
.hljs-bullet,
.hljs-addition {
	color: #718c00;
}

/* Tomorrow Blue */
.hljs-title,
.hljs-section {
	color: #4271ae;
}

/* Tomorrow Purple */
.hljs-keyword,
.hljs-selector-tag {
	color: #8959a8;
}

.hljs {
	display: block;
	overflow-x: auto;
	color: #4d4d4c;
	padding: 0.5em;
}

.hljs-emphasis {
	font-style: italic;
}

.hljs-strong {
	font-weight: bold;
}
</style>

<style>
/*
 * Markdown PDF CSS
 */

 body {
	font-family: -apple-system, BlinkMacSystemFont, "Segoe WPC", "Segoe UI", "Ubuntu", "Droid Sans", sans-serif, "Meiryo";
	padding: 0 12px;
}

pre {
	background-color: #f8f8f8;
	border: 1px solid #cccccc;
	border-radius: 3px;
	overflow-x: auto;
	white-space: pre-wrap;
	overflow-wrap: break-word;
}

pre:not(.hljs) {
	padding: 23px;
	line-height: 19px;
}

blockquote {
	background: rgba(127, 127, 127, 0.1);
	border-color: rgba(0, 122, 204, 0.5);
}

.emoji {
	height: 1.4em;
}

code {
	font-size: 14px;
	line-height: 19px;
}

/* for inline code */
:not(pre):not(.hljs) > code {
	color: #C9AE75; /* Change the old color so it seems less like an error */
	font-size: inherit;
}

/* Page Break : use <div class="page"/> to insert page break
-------------------------------------------------------- */
.page {
	page-break-after: always;
}

</style>

<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>
</head>
<body>
  <script>
    mermaid.initialize({
      startOnLoad: true,
      theme: document.body.classList.contains('vscode-dark') || document.body.classList.contains('vscode-high-contrast')
          ? 'dark'
          : 'default'
    });
  </script>
<h1 id="week-8-practical---vcf-analysis---dr-rick-tearle">Week 8 Practical - VCF Analysis - Dr Rick Tearle</h1>
<h1 id="vcf-analysis-practical">VCF Analysis practical</h1>
<p>As with all previous practicals, start by creating a <code>~/Practical_8/</code> subdirectory on your VM. You will need to move the data and R script and Rproj files (see below) from your <code>~/data/variant_calling/</code> directory to the <code>~/Practical_8/</code> directory prior to working on it.</p>
<h2 id="about-this-practical">About this practical</h2>
<p>This practical is designed to give you experience with VCF files and their analysis and will use the genome and NGS data from Assignment 4. It includes an Rscript and BAM and VCF files based on that data.</p>
<p>The files you will need are in your <code>~/sudent/data/variant_calling</code> directory. These are the files you should have:</p>
<pre class="hljs"><code><div>-rw-r--r-- 1 student student 9.1M Sep 12 06:07 Arabidopsis_thaliana.TAIR10.48.gff3.gz
-rw-r--r-- 1 student student  35M Sep 12 06:07 Arabidopsis_thaliana.TAIR10.dna.toplevel.fa.gz
-rw-rw-r-- 1 student student 1.1G Sep 12 06:27 SRR5882792_Athaliana_TAIR10.bam
-rw-rw-r-- 1 student student 351K Sep 14 01:17 SRR5882792_Athaliana_TAIR10.bam.bai
-rw-rw-r-- 1 student student  15M Sep 12 06:28 SRR5882792_Athaliana_TAIR10_picard.vcf
-rwxr-xr-x 1 student student 9.3K Sep 14 05:47 VCF_Analysis.R
-rwxr-xr-x 1 student student  234 Sep 14 05:47 VCF_Analysis.Rproj

</div></code></pre>
<p>Move these files to your <code>~/Practical_8/</code> directory. You will then be able to open the Rproj and Rscript files to start working.</p>
<p>The information for reads containing variants is stored in the BAM files and variant information is stored in the VCF files. We will visualise both to see the evidence that supports the variant calls. You will use an R script to summarise the data in the VCF file and to identify interesting variants to interrogate further.</p>
<p>For the visualisation of variant data you will use IGV <a href="https://software.broadinstitute.org/software/igv/">Integrated Genome Viewer</a> and you will need to install this on your own computer as it will not run on a browser from your VM. There are versions of IGV available for download for any operating system you might be using: Windowns, MacOS or Linux. Visualisation of your results will require you to download the R output of the analysis to your computer.</p>
<p>Specific instructions will be provided during the practical to allow you to carry out the steps required. In previous practicals you have been able to access background material, along with a detailed description of practical steps and the reasons behind them. For this practical, all of that information will be provided verbally during the scheduled practical session, and if time permits it will be incorporated into a revised version of this document. In contrast to previous practical sessions, we will record this practical and upload it to MyUni to provide an accessible record of the instructions and procedures.</p>
<h2 id="set-up">Set up</h2>
<p>First load the packages we need.</p>
<pre class="hljs"><code><div>library(magrittr)
library(vcfR)
library(tidyverse)
</div></code></pre>
<p>Then load the functions we need.</p>
<pre class="hljs"><code><div>'%nin%' &lt;- Negate('%in%')

ulength &lt;- function(x) {x %&gt;% unique() %&gt;% length()}
fsummary &lt;- function(x) {x %&gt;% as.factor() %&gt;% summary()}
</div></code></pre>
<p>Now set up th paths and environment for input and output.</p>
<pre class="hljs"><code><div>DirIn &lt;- &quot;/home/student/Practical_8&quot;
DirOut &lt;- &quot;/home/student/Practical_8/Out&quot;
if(! dir.exists(DirOut)) {dir.create(DirOut)}

DirPlot &lt;- file.path(DirOut, &quot;Plots&quot;)
if(! dir.exists(DirPlot)) {dir.create(DirPlot)}
</div></code></pre>
<h2 id="some-notes">Some notes</h2>
<p>The R package vcfR contains code to load and manipulate VCF files. We will only use a small proportion of its functionality but if you find yourself interrogating VCF files regularly, it will be worth your while reading up on the functions in this package.</p>
<p>I have provided code to load reference sequences and annotation (gff) files in R as well, see sections <strong>Load GFF File</strong> and ##Load Reference File##. We will not use them here: they are provided for the sake of completeness.</p>
<p>After loading packages, we will start with the section <strong>IGV</strong>. This just contains a lists of co-ordinates for you to look at in IGV. We will inspect through these locations and I will have some comments on each.</p>
<p>Next, go to <strong>Load VCF</strong>. After loading the VCF file we will look at how to access some of the data.</p>
<p>The next section is <strong>Analyse Genotypes</strong>. We will extract a subset of the columns from the VCF file as a data frame and reformat some of them. Data frames are a convenient way to store and interrogate this information.</p>
<p>We will dig in to the data to see what evidence is being used to call variants, in <strong>Plots</strong>. We will look at the relationship between the genotypes called, the number of reads and the strength of evidence. When we find an interesting relationship we will take examples and look at them in IGV, where we can get a sense of the reads that support a call.</p>
<p>Finally, I would like you to take the data and the code, and generate your own plots etc, to see if you can find a threshold to balance True and False Positives. The columns Qual and GLB will be useful here.</p>

</body>
</html>
